rechunker/api.py

```diff
@@ -10,7 +10,41 @@
 from rechunker.algorithm import rechunking_plan
 
 
-def rechunk_zarr2zarr_w_dask(
+def _shape_dict_to_tuple(dims, shape_dict):
+    # convert a dict of shape
+    shape = [shape_dict[dim] for dim in dims]
+    return tuple(shape)
+
+
+def _get_dims_from_zarr_array(z_array):
+    # use Xarray convention
+    # http://xarray.pydata.org/en/stable/internals.html#zarr-encoding-specification
+    return z_array.attrs["_ARRAY_DIMENSIONS"]
+
+
+def rechunk(
+    source_array,
+    target_chunks,
+    max_mem,
+    target_store,
+    temp_store=None,
+    source_storage_options={},
```

Comment:
nitpick: make these default to None to avoid a mutable default and check

```python
source_storage_options = source_storage_options or {}
...
```